lib/bpf/bpf.go

```diff
@@ -130,11 +130,6 @@ func New(config *servicecfg.BPFConfig) (BPF, error) {
 		return &NOP{}, nil
 	}
 
-	// Check if the host can run BPF programs.
-	if err := IsHostCompatible(); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
 	// Create a cgroup controller to add/remote cgroups.
 	cgroup, err := controlgroup.New(&controlgroup.Config{
```

Comment:
As I understand since we don't have this check, and we create cgroup and then fail, we have to close it in error
```golang
	// Create a cgroup controller to add/remote cgroups.
	s.cgroup, err = controlgroup.New(&controlgroup.Config{
		MountPath: config.CgroupPath,
		RootPath:  config.RootPath,
	})
	if err != nil {
		return nil, trace.Wrap(err)
	}
	defer func() {
		if err != nil {
			if err := s.cgroup.Close(true); err != nil {
				log.WithError(err).Warn("Failed to close cgroup")
			}
		}
	}()
```

Reply:
Yes, we need to clean the cgroups if we fail